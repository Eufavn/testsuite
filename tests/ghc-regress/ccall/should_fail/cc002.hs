-- !!! cc002 -- ccall with non-standard boxed arguments and results

module Test where

import Foreign

-- Test returning results

a :: IO ForeignObj
a = _ccall_ a

b :: IO (StablePtr Double)
b = _ccall_ b

-- Test taking arguments

c :: ForeignObj -> IO Int
c x = _ccall_ c x

d :: StablePtr Int -> IO Int
d x = _ccall_ d x
