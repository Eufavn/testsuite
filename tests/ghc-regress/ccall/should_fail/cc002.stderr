
cc002.hs:10:
    Could not deduce (GHC.Prim.CReturnable (ForeignPtr a))
	from the context ()
    Probable fix:
	Add (GHC.Prim.CReturnable (ForeignPtr a))
	to the type signature(s) for `a'
	Or add an instance declaration for (GHC.Prim.CReturnable (ForeignPtr
								      a))
    arising from the result of the _ccall_ to `a' at cc002.hs:10
    In the definition of `a': _ccall_ a
